### DISCUSSION TITLE
Size of the keys

### DISCUSSION THREAD

**Discussion by vinschmid** [jazz-soft/JZZ-input-Kbd#8] [Q&A]
Is there a possibility to set the size of the keys (bl, wl, bw, ww) dynamically instead of px?

**Comment by jazz-soft** [comment]
Hi! Thanks for asking!
What is your use case? There are some options for responsive design.
Or, you can create a new keyboard each time you want to change the size.

**Reply by vinschmid** [reply]
I want the keyboard to be the full width of the window (and white keys should be 10% of the height). No matter what size the screen (or better the window) has. It is for a software that will be a website someday...

**Comment by jazz-soft** [comment]
There is no easy way to do it at the moment.
You may want to try an SVG example at https://jazz-soft.net/demo/PianoStyle.html
But that sounds like a nice new feature to implement in the future version.

**Comment by jazz-soft** [comment]
Please check the latest release.
It now accepts key height/width in the percentage of the outer container, e.g: ww:'10%', bw:'8%' ...

**Reply by vinschmid** [reply]
Thank you for this update! Is there anything specific to keep in mind when adjusting the width of the black keys? So far, I have managed to make the white keys adjust to the screen as I want. In my case, I wrote "ww: '2.31%'". For the black keys, I take 2/3 of that, so "bw: '1.54%'". However, the positioning of the black keys is very strange - sometimes too far left, sometimes too far right, never exactly centered between the corresponding white keys. Therefore, I am wondering if there is something specific to consider. Once again, thank you very much for your efforts!

**Reply by vinschmid** [reply]
<img width="960" alt="Screenshot 2024-07-09 182350" src="https://github.com/jazz-soft/JZZ-input-Kbd/assets/172148021/850a9f6b-68ec-417c-8f36-c75b03f238a8">

Here is a picture of what it looks like...

**Comment by jazz-soft** [comment]
That does not look correct...
Can you construct a minimal example with such a behavior?
I would look at it in the debugger...
